kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -296,4 +310,50 @@ private Optional<Long> loadLatestTransactionVersion(Engine engine, String applic
 
     return Optional.empty();
   }
+
+  /**
+   * Retrieves a map of domainName to {@link DomainMetadata} from the log files.
+   *
+   * <p>Now loading domain metadata requires an additional round of log replay so this is done
```

Comment:
we should do this during the P & M log replay

Reply:
According to our discussion, we will address this in a separate PR.